### PR TITLE
Combine settings for navigation action after removing a message

### PR DESCRIPTION
**Pull request by cketti**
This combines the settings "Return to list after delete" and "Show next message after delete" into a single setting "After deleting or moving a message" (suggestions for a better name are welcome).

After the user deletes or moves a message in the message view screen, the available navigation options are:
- Return to message list
- Show previous message
- Show next message

We get rid of the "magic" mode that tries to keep going in the direction in which the user was moving previously.

### CODE REVIEW

**Review by wmontwe — CHANGES_REQUESTED** [review]
I like that the navigation flow logic has been streamlined. While the name `PostRemoveNavigation` isn't ideal, I can't provide a better one.

Some of the modified classes reached a size that harms readability, and adding more code to them would be counterproductive.

Optional: The migration logic is scattered across various modules and classes, making it easy to overlook a step. This area could definitely improve with some refactoring.

File: app/core/src/main/java/com/fsck/k9/preferences/GeneralSettingsDescriptions.java
[the remark this answers is not in the document]
Comment (cketti) [reply app/core/src/main/java/com/fsck/k9/preferences/GeneralSettingsDescriptions.java:1]: The `SettingsUpgrader` implementations have been extracted to separate files in #7220.

File: app/core/src/main/java/com/fsck/k9/K9.kt
[the remark this answers is not in the document]
Comment (cketti) [reply app/core/src/main/java/com/fsck/k9/K9.kt:1]: I'd like to defer this until we have figured out a way how settings can be part of a feature (module).

[the remark this answers is not in the document]
Comment (wmontwe) [reply app/core/src/main/java/com/fsck/k9/K9.kt:1]: Yeah, this is a tricky beast, let's do it later.

File: app/core/src/main/java/com/fsck/k9/preferences/GeneralSettingsDescriptions.java
[the remark this answers is not in the document]
Comment (wmontwe) [reply app/core/src/main/java/com/fsck/k9/preferences/GeneralSettingsDescriptions.java:1]: Nice

File: app/core/src/main/java/com/fsck/k9/preferences/upgrader/GeneralSettingsUpgraderTo89.kt
Comment (wmontwe) [inline app/core/src/main/java/com/fsck/k9/preferences/upgrader/GeneralSettingsUpgraderTo89.kt:1]: 👍

**Review by wmontwe — APPROVED** [review]
👍

**Pull request merged**